src/plugins/discover/public/application/main/hooks/use_discover_state.ts

```diff
@@ -226,6 +249,41 @@ export function useDiscoverState({
     }
   }, [initialFetchStatus, refetch$, indexPattern, savedSearch.id]);
 
+  const fetchResults = useCallback(() => {
```

Comment:
Nit: I think this name might be misleading, maybe `getResultColumns` might be more descriptive?